frontend/src/components/secret/Details.tsx

```diff
@@ -14,11 +17,35 @@ export default function SecretDetails() {
       namespace={namespace}
       withEvents
       extraInfo={item =>
-        item &&
-        _.map(item.data, (value, key) => ({
-          name: key,
-          value: <SecretField value={value} />,
-        }))
+        item && [
+          {
+            name: t('translation|Type'),
+            value: item.type,
+          },
+        ]
+      }
+      extraSections={item =>
+        item && [
+          {
+            id: 'headlamp.secrets-data',
+            section: () => {
+              const itemData = item?.data || {};
+              const mainRows = Object.entries(itemData).map((item: unknown[]) => ({
+                name: item[0],
+                value: <SecretField value={item[1]} />,
+              })) as NameValueTableRow[];
```

Comment:
If you declare the mainRows as `const mainRows: NameValueTableRow[]`, that should still get the right type while not masking any possible mistypes in the actual data.

Reply:
This has not been addressed (same for the similar change in the next commit).

Reply:
Oh it seems i reverted this.